modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2289 @@
+  {
+    const std::string en_kernel_type = "FVTimeKernel";
+    InputParameters params = _factory.getValidParams(en_kernel_type);
+    params.set<std::vector<SubdomainName>>("block") = _blocks;
+    params.set<NonlinearVariableName>("variable") = vname;
+
+    _problem->addFVKernel(en_kernel_type, "ins_" + vname + "_time", params);
+  }
+}
+
+void
+NSFVAction::addINSMassKernels()
+{
+  if (_porous_medium_treatment)
+  {
+    const std::string kernel_type = "PINSFVMassAdvection";
+    InputParameters params = _factory.getValidParams(kernel_type);
+    params.set<std::vector<SubdomainName>>("block") = _blocks;
+    params.set<NonlinearVariableName>("variable") = NS::pressure;
+    params.set<MooseFunctorName>(NS::density) = _density_name;
+    params.set<MooseEnum>("velocity_interp_method") = "rc";
+    params.set<UserObjectName>("rhie_chow_user_object") = "pins_rhie_chow_interpolator";
+    params.set<MooseEnum>("advected_interp_method") = _mass_advection_interpolation;
+
+    _problem->addFVKernel(kernel_type, "pins_mass_advection", params);
+  }
+  else
+  {
+    const std::string kernel_type = "INSFVMassAdvection";
+    InputParameters params = _factory.getValidParams(kernel_type);
+    params.set<std::vector<SubdomainName>>("block") = _blocks;
+    params.set<NonlinearVariableName>("variable") = NS::pressure;
+    params.set<MooseFunctorName>(NS::density) = _density_name;
+    params.set<MooseEnum>("velocity_interp_method") = "rc";
+    params.set<UserObjectName>("rhie_chow_user_object") = "ins_rhie_chow_interpolator";
+    params.set<MooseEnum>("advected_interp_method") = _mass_advection_interpolation;
+
+    _problem->addFVKernel(kernel_type, "ins_mass_advection", params);
+  }
+
+  if (getParam<bool>("pin_pressure"))
+  {
+    MooseEnum pin_type = getParam<MooseEnum>("pinned_pressure_type");
+    std::string kernel_type;
+    if (pin_type == "point-value")
+      kernel_type = "FVPointValueConstraint";
+    else
+      kernel_type = "FVIntegralValueConstraint";
+    InputParameters params = _factory.getValidParams(kernel_type);
+    params.set<CoupledName>("lambda") = {"lambda"};
+    params.set<Real>("phi0") = getParam<Real>("pinned_pressure_value");
+    params.set<NonlinearVariableName>("variable") = NS::pressure;
+    if (pin_type == "point-value")
+      params.set<Point>("point") = getParam<Point>("pinned_pressure_point");
+
+    _problem->addFVKernel(kernel_type, "ins_mass_pressure_constraint", params);
+  }
+}
+
+void
+NSFVAction::addINSMomentumAdvectionKernels()
+{
+  if (_porous_medium_treatment)
+  {
+    const std::string adv_kernel_type = "PINSFVMomentumAdvection";
+    InputParameters params = _factory.getValidParams(adv_kernel_type);
+    params.set<std::vector<SubdomainName>>("block") = _blocks;
+    params.set<MooseFunctorName>(NS::density) = _density_name;
+    params.set<MooseFunctorName>(NS::porosity) = _porosity_name;
+    params.set<MooseEnum>("velocity_interp_method") = "rc";
```

Comment:
umm I understand why but also this is confusing with the parameter face_interpolation for velocity

Reply:
right now we only allow RC interpolation for the face flux. I'll add this to the documentation and extend the strings used for the parameter descriptions.